File: c_src/write_string.c
// Linux
#define _GNU_SOURCE

#include <stdio.h>
#include <sys/mman.h>
#include <unistd.h>
#include <string.h>

int main()
{
    size_t size = getpagesize();

    void *ptr = mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANON, -1, 0);

    if (ptr == MAP_FAILED)
    {
        perror("mmap");
        return 1;
    }

    const char *message = "Hello, World!";
    memcpy(ptr, message, strlen(message) + 1);

    printf("%s\n", (char *)ptr);

    munmap(ptr, size);

    return 0;
}

File: examples/write_string.rs
use memmap2::MmapMut;
use std::str;

fn main() {
    let mut mmap = MmapMut::map_anon(4096).expect("mmap");

    let message = b"Hello, World!";

    mmap[..message.len()].copy_from_slice(message);

    let text = str::from_utf8(&mmap[..message.len()]).expect("utf8");

    println!("{}", text);
}
